aztokenprovider/token_provider.go

```diff
@@ -106,6 +106,9 @@ func NewAzureAccessTokenProvider(settings *azsettings.AzureSettings, credentials
 			usernameAssertion: tokenEndpoint.UsernameAssertion,
 			tokenRetriever:    tokenRetriever,
 		}, nil
+	case *azcredentials.AzureClientPasswordCredentials:
```

Comment:
```suggestion
	case *azcredentials.AzureEntraPasswordCredentials:
```

Incase we do decide to keep this 😊